app/controllers/papers_controller.rb

```diff
@@ -196,7 +196,10 @@ def show
       head 404 and return unless can_see_hidden_paper?(@paper)
     end
 
-    render layout: false
+    respond_to do |format|
+      format.pdf { redirect_to @paper.pdf_url, :status => 301 }
```

Comment:
If format is `pdf` we simply redirect to the existing `pdf_url` on GitHub pages.